### ISSUE TITLE
SDL2 2.0.9 causes "sdl2.ext.common.SDLError: b'No available video device'"

### ISSUE THREAD

**Issue by djhoese** [conda-forge/sdl2-feedstock#13]
Vispy uses the pysdl package and ever singe the sdl 2.0.9 package update here I've been getting:

```
sdl2.ext.common.SDLError: b'No available video device'
```

I'm still debugging but here is a previously working job:

https://api.travis-ci.org/v3/job/523227462/log.txt

And with the new package and failing:

https://api.travis-ci.org/v3/job/525753657/log.txt

This could be travis's fault or vispy's use of SDL but I wouldn't have expected a change from 2.0.8 to 2.0.9 to break our usage.

CC @larsoner @kmuehlbauer

**Comment by djhoese** [comment]
Here's the non-txt log for the failing job: https://travis-ci.org/vispy/vispy/jobs/525753657

**Comment by djhoese** [comment]
Forcing to 2.0.8 now passes: https://travis-ci.org/vispy/vispy/jobs/525766502

**Comment by matham** [comment]
That is strange, looking at the difference in your two builds, it is only differs in the sdl2 version AFAICS. Yet in the [PR for sdl2 2.0.9](https://github.com/conda-forge/sdl2-feedstock/pull/12/files#diff-e178b687b10a71a3348107ae3154e44c) there was basically no difference between with master other than the sdl2 source code.

I wonder if this has something to do with the recent change in conda to make [libgl part of the cdt](https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl). Maybe we now have to explicitly list those gl dependencies in the recipe. Another difference is that the previous build was on travis, while the current one was on azure.

Looking at the most recent [build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=30601&view=logs&jobId=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&taskId=c09a5a47-eec3-5700-9284-6f1bd7b23501&lineStart=520&lineEnd=521&colStart=1&colEnd=1), it says

```
checking for X... no
checking for OpenGL support... no
checking that generated files are newer than configure... done
```

while the [build log](https://circleci.com/gh/conda-forge/sdl2-feedstock/42?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) for the sdl2 2.0.8 says

```
checking for XF86VidModeQueryVersion in -lXxf86vm... no
checking for OpenGL (GLX) support... yes
checking for EGL support... yes
checking for OpenGL ES v1 headers... no
checking for OpenGL ES v2 headers... yes
checking for pkg-config... (cached) /usr/bin/pkg-config
```

I just made a [PR](https://github.com/conda-forge/sdl2-feedstock/pull/14) that hopefully fixes this. Lets see if it build with gl support turned ON for sdl2.

**Comment by matham** [comment]
The PR does indeed fix the issue. The [log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=31459&view=logs&jobId=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&taskId=c09a5a47-eec3-5700-9284-6f1bd7b23501&lineStart=744&lineEnd=745&colStart=1&colEnd=1) now says:

```
checking for X... no
checking for OpenGL (GLX) support... yes
checking for EGL support... yes
checking for OpenGL ES v1 headers... no
checking for OpenGL ES v2 headers... yes
checking for pkg-config... (cached) /usr/bin/pkg-config
```

**Comment by djhoese** [comment]
This still seems to be failing on this travis job: https://travis-ci.org/vispy/vispy/jobs/525969469

Note that we are also installing SDL at the system level, but from what I can tell this isn't being used and we've been doing this the entire time.

**Comment by matham** [comment]
I think this may be a problem with your usage of travis (as it's being compiled now on azure, not travis) or more likely the runtime configuration. Our recipe and glew list the same dependencies, which makes me think we are compiling correctly, because glew needs gl.

Glew does list some test dependencies during [test](https://github.com/conda-forge/glew-feedstock/blob/master/recipe/meta.yaml#L43). Perhaps try installing them.

**Comment by kmuehlbauer** [comment]
On line 813 in the latest [azure logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=31477&view=logs) the X11 video driver is missing. (It is available in the 2.0.8 [Circle builds](https://circleci.com/gh/conda-forge/sdl2-feedstock/43?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)). It seems, that the X11 drivers are thus not compiled in the current `sdl2`, leading to the mentioned problems.

To further dig into this I also compiled a little C++ program (from here https://github.com/xyproto/sdl2-examples) which works fine with 2.0.8 but errors out using 2.0.9. So this is no `vispy`-issue and no `pysdl2`-issue. I really appreciate any further guidance on how to re-activate X11 drivers for `sdl2` builds.

**Comment by kmuehlbauer** [comment]
@dschreij @matham Thanks for fixing!
